spec/controllers/products_controller_spec.rb

```diff
@@ -0,0 +1,73 @@
+require 'rails_helper'
+
+RSpec.describe ProductsController, type: :controller do
+
+  describe "Products controller" do
+    login_user
```

Comment:
All the tests are for signed in users. How about when they are not logged in? Need tests that will assert to fail for non-signed in users. Like if they try to go to edit page. Then also tests for someone trying to edit someone else's product, etc.